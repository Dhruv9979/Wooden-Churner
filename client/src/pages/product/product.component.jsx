import React, { useState } from 'react';
import { connect } from 'react-redux';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faArrowLeft } from '@fortawesome/free-solid-svg-icons';

import CollectionItem from '../../components/collection-item/collection-item.component';
import ProductInformation from '../../components/product-information/product-information.component';
import Modal from '../../components/modal/modal.component';

import { selectProducts } from '../../redux/shop/shop.selectors';

import {
    ProductPageContainer,
    ProductTitle,
    ProductItemsContainer,
    OptionButton,
} from './product.styles';

export const ProductPage = ({ items, history }) => {
    const { name, products, description } = items;

    const [modalState, setModalState] = useState(false);

    const toggleModalState = () => {
        setModalState(!modalState);
    };

    return (
        <ProductPageContainer>
            <Modal
                modalState={modalState}
                toggleModalState={toggleModalState}
                form={false}
                header="Please Sign In/Sign Up by visiting to Sign in Page to add items to cart."
            />
            <OptionButton onClick={() => history.goBack()}>
                <FontAwesomeIcon
                    icon={faArrowLeft}
                    style={{ marginRight: '10px' }}
                />
                GO BACK
            </OptionButton>
            <ProductTitle>{name.toUpperCase()}</ProductTitle>
            {description && (
                <ProductInformation description={description} name={name} />
            )}
            <ProductItemsContainer>
                {Object.entries(products).map(([key, val]) => (
                    <CollectionItem
                        key={val.id}
                        item={val}
                        toggleModalState={toggleModalState}
                    />
                ))}
            </ProductItemsContainer>
            
        </ProductPageContainer>
    );
};

const mapStateToProps = (state, ownProps) => ({
    items: selectProducts(
        ownProps.match.params.collectionId,
        ownProps.match.params.productId
    )(state),
});

export default connect(mapStateToProps)(ProductPage);
