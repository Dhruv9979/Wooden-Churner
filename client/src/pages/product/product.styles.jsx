import styled from 'styled-components';

export const ProductPageContainer = styled.div`
    display: flex;
    flex-direction: column;
`;

export const ProductTitle = styled.h2`
    font-size: 38px;
    margin: 0 auto 30px;
    color: #762522;
`;

export const ProductItemsContainer = styled.div`
    display: grid;
    grid-template-columns: 1fr 1fr 1fr 1fr;
    grid-gap: 10px;

    & > div {
        margin-bottom: 30px;
    }

    @media screen and (max-width: 800px) {
        grid-template-columns: 1fr 1fr;
        grid-gap: 15px;
    }
`;

export const OptionButton = styled.button`
    padding: 10px 15px 50px;
    width: 150px;
    background-color: #fff;
    border: none;
    text-align: center;
    text-decoration: none;
    font-size: 20px;
    margin: 4px 2px;
    cursor: pointer;
    outline: none;
    color: #762522;
    font-family: 'Poppins', serif;
`;
