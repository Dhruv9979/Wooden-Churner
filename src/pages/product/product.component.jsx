import React from "react";
import { connect } from "react-redux";

import CollectionItem from "../../components/collection-item/collection-item.component";
// import ProductInformation  from '../../components/product-information/product-information.component';

import { selectProduct } from "../../redux/collection/collection.selectors";

import {
	ProductPageContainer,
	ProductTitle,
	ProductItemsContainer,
} from "./product.styles";

const ProductPage = ({items}) => {
	const { name, products } = items;
    return (
		<ProductPageContainer>
			<ProductTitle>{name}</ProductTitle>
			{/* {Object.entries(description).map(([key, val]) => 
				<ProductInformation h2 = {h2} p = {val.p} li = {val.li} />
			)} */}
			{/* <ProductInformation description={description} /> */}
			<ProductItemsContainer>
				{products.map((product) => (
					<CollectionItem key={product.id} item={product} />
				))}
			</ProductItemsContainer>
		</ProductPageContainer>
	);
};

const mapStateToProps = (state, ownProps) => ({
	items: selectProduct(ownProps.match.params.productId)(state),
});

export default connect(mapStateToProps)(ProductPage);