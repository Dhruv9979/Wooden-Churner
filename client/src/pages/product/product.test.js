import React from "react";
import { shallow } from "enzyme";

import { ProductPage } from "./product.component";
import CollectionItem from "../../components/collection-item/collection-item.component";

describe('Product Page', () => {
    let wrapper;
    let mockItems = {
        flours: { flour1: { id: 1 }, flour2: { id: 2 }, flour3: { id: 3 } },
    };
    
    beforeEach(() => {
        const mockProducts = {
            products: mockItems,
            name: "Flours",
        };
        
        wrapper = shallow(<ProductPage items={mockProducts} />);
    });
    
    it("should render the Product Page", () => {
        expect(wrapper).toMatchSnapshot();
    });
    
    it("should render the same number of Products as product object", () => {
        expect(wrapper.find(CollectionItem).length).toBe(Object.keys(mockItems).length);
    });
});
