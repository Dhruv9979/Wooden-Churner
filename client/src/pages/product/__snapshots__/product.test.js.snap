// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product Page should render the Product Page 1`] = `ShallowWrapper {}`;
